app/src/main/java/com/example/movieapp/ui/viewmodel/MainViewModel.kt

```diff
@@ -51,11 +55,17 @@ class MainViewModel : ViewModel() {
 
     val moviesViewHolderListener = object : MovieViewHolder.MoviesViewHolderListener {
         override fun onItemLiked(movie: Movie) {
-            if (likedList.value?.contains(movie) == true) {
-                deleteMovie(movie)
-            } else {
-                insertMovie(movie)
-            }
+            doOnLikePress(movie)
+        }
+
+        override fun onDetailsRequested(
+            view: View,
+            movie: Movie
+        ) {
+            Log.d(TAG, "onDetailsRequested: image is clicked")
+            val action =
+                MainFragmentDirections.actionMainFragmentToDetailFragment(movie.id.toString())
```

Comment:
Navigation should be done from the fragment, plus, why do you convert your int into a string then in your DetailViewModel convert your string into a int 🧐